src/Security/Authorization/Policy/src/AuthorizationMiddlewareOptions.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Authorization
+{
+    /// <summary>
+    /// Provides programmatic configuration used by the <see cref="AuthorizationMiddleware"/>.
+    /// </summary>
+    public class AuthorizationMiddlewareOptions
+    {
+        /// <summary>
+        /// Determines whether HttpContext will be passed as the resource to authorization instead of the matching endpoint.
+        /// Defaults to false.
+        /// </summary>
+        public bool UseHttpContextAsResource { get; set; } = false;
```

Comment:
Mvc uses the terms `Allow` and `Suppress` for these sort of switches. It would be good to continue using the pattern. 

```suggestion
        public bool AllowUseHttpContextAsAuthorizationResource { get; set; }
```

Having this be an `HttpContext` instance is closer to the MVC's behavior of using an `ActionContext` as the resource in auth filter which is why I don't feel too bad about having this be on by default. If we do this, the preferred prefix is Suppress:

```suggestion
        public bool SuppressUseHttpContextAsAuthorizationResource { get; set; }
```